haigha/classes/basic_class.py

```diff
@@ -53,8 +54,30 @@ def _cleanup(self):
         self._get_cb = None
         self._recover_cb = None
         self._cancel_cb = None
+        self._return_listener = None
```

Comment:
Hi @awestendorf, referring to the preceding comment, in order to support the non-delivery-confirmation use case, we still need to allow the user to register a callback to listen on Basic.Return. And since there is no deterministic mechanism to determine when Basic.Return will or will not be returned, I think that a channel-global Basic.Return listener may be our only option.

A channel-global Basic.Return listener as I implemented in this PR would address the needs of non-delivery-confirmation use cases; and it would also work for the delivery-confirmation use cases along with the present channel-global ack/nack listeners. I think this will work out well and is consistent with the current ack/nack API. What do you think?

If you like the idea that I described in the previous comment for "puback_cb", then we can provide that enhanced functionality via separate PR without any damage to existing users.